File: mockez-back/src/main/java/com/mockez/domain/model/entity/composite_key/GroupAccessPK.java
package com.mockez.domain.model.entity.composite_key;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.io.Serializable;
import java.util.UUID;

@Data
@Embeddable
public class GroupAccessPK implements Serializable {

    @Column(nullable = false)
    private UUID groupId;

    @Column(nullable = false)
    private UUID userId;
}

File: mockez-front/src/app/core/model/composite_key/group-access-pk.ts
export class GroupAccessPK {

  private _groupId: string;
  private _userId: string;

  constructor(groupId: string, userId: string) {
    this._groupId = groupId;
    this._userId = userId;
  }

  get groupId(): string {
    return this._groupId;
  }

  set groupId(value: string) {
    this._groupId = value;
  }

  get userId(): string {
    return this._userId;
  }

  set userId(value: string) {
    this._userId = value;
  }
}
